protostar/cairo_testing/cairo_test_execution_state.py

```diff
@@ -0,0 +1,81 @@
+import dataclasses
+from copy import deepcopy
+from dataclasses import dataclass
+
+from starkware.crypto.signature.fast_pedersen_hash import pedersen_hash_func
+from starkware.starknet.business_logic.fact_state.patricia_state import (
+    PatriciaStateReader,
+)
+from starkware.starknet.business_logic.fact_state.state import SharedState
+from starkware.starknet.business_logic.state.state_api_objects import BlockInfo
+from starkware.starknet.definitions.general_config import StarknetGeneralConfig
+from starkware.starknet.testing.starknet import Starknet
+from starkware.starknet.testing.state import StarknetState
+from starkware.storage.dict_storage import DictStorage
+from starkware.storage.storage import FactFetchingContext
+from typing_extensions import Self
+
+from protostar.compiler import ProjectCompiler
+from protostar.cheatable_starknet.cheatable_cached_state import (
+    CheatableCachedState,
+)
+from protostar.testing.stopwatch import Stopwatch
+from protostar.testing.test_config import TestConfig
+from protostar.testing.test_context import TestContext
+from protostar.testing.test_output_recorder import OutputRecorder
+
+
+@dataclass
+class CairoTestExecutionState:
+    starknet: Starknet
```

Comment:
I think it would be beneficial to use CheatableCahedState here, but it is not something to address in this PR